src/contexts/LoginContext.tsx

```diff
@@ -1,41 +1,67 @@
 import axios from "axios"
 import { isEmpty, omitBy } from "lodash"
-import { createContext, useEffect, useContext } from "react"
+import {
+  createContext,
+  useEffect,
+  useContext,
+  PropsWithChildren,
+  useCallback,
+} from "react"
 
 import { LOCAL_STORAGE_KEYS } from "constants/localStorage"
 
 import { useLocalStorage } from "hooks/useLocalStorage"
 
 const { REACT_APP_BACKEND_URL: BACKEND_URL } = process.env
 
-const LoginContext = createContext(null)
+interface LoginContextProps {
+  userId: string
+  email: string
+  contactNumber: string
+  logout: () => Promise<void>
+  verifyLoginAndSetLocalStorage: () => Promise<void>
+}
+
+interface loggedInUserProps {
+  userId: string
+  email: string
+  contactNumber: string
+}
+
+const LoginContext = createContext<null | LoginContextProps>(null)
 
-const LoginConsumer = ({ children }) => {
+const useLoginContext = (): LoginContextProps => {
   const loginContextData = useContext(LoginContext)
   if (!loginContextData)
     throw new Error("useLoginContext must be used within an LoginProvider")
 
-  return <LoginContext.Consumer>{children}</LoginContext.Consumer>
+  return loginContextData
 }
 
-const LoginProvider = ({ children }) => {
+const LoginProvider = ({
+  children,
+}: PropsWithChildren<Record<string, never>>): JSX.Element => {
   const [storedUserId, setStoredUserId, removeStoredUserId] = useLocalStorage(
     LOCAL_STORAGE_KEYS.GithubId,
     "Unknown user"
   )
-  const [storedUser, setStoredUser, removeStoredUser] = useLocalStorage(
-    LOCAL_STORAGE_KEYS.User,
-    {}
-  )
+  const [
+    storedUser,
+    setStoredUser,
+    removeStoredUser,
+  ] = useLocalStorage(LOCAL_STORAGE_KEYS.User, { email: "", contactNumber: "" })
 
-  const [, , removeSites] = useLocalStorage(LOCAL_STORAGE_KEYS.SitesIsPrivate)
-  const verifyLoginAndSetLocalStorage = async () => {
+  const [, , removeSites] = useLocalStorage(
+    LOCAL_STORAGE_KEYS.SitesIsPrivate,
+    false
+  )
+  const verifyLoginAndSetLocalStorage = useCallback(async () => {
     const { data } = await axios.get(`${BACKEND_URL}/auth/whoami`)
     const loggedInUser = omitBy(data, isEmpty)
 
     setStoredUserId(loggedInUser.userId)
-    setStoredUser(loggedInUser)
-  }
+    setStoredUser((loggedInUser as unknown) as loggedInUserProps)
```

Comment:
we can avoid the cast by setting the generic type on `axios.get` like so: `axios.get<LoggedInUser>`

Reply:
This was actually an issue regardless because of `omitBy` returning a `Partial<object>` - it's been removed in https://github.com/isomerpages/isomercms-frontend/pull/959/commits/021633d04e612010214a625aef12b6522edfa527 so we no longer need to cast after the above suggestion